### PR TITLE
Implement WebviewPanel.reveal()

### PR DESCRIPTION
**Pull request by vparfonov**
Need for #4339

### CODE REVIEW

**Comment by benoitf** [comment]
@vparfonov FYI the Signed-Off is invalid (eclipsefdn/eca) check

File: packages/plugin-ext/src/main/browser/webviews-main.ts [hunk 1]
```diff
@@ -98,7 +98,14 @@ export class WebviewsMainImpl implements WebviewsMain {
...
     }
     $reveal(handle: string, showOptions: WebviewPanelShowOptions): void {
-        throw new Error('Method not implemented.');
+        const webview = this.getWebview(handle);
+        if (webview.isDisposed) {
+            return;
+        }
+        this.shell.revealWidget(webview.id);
+        if (showOptions.preserveFocus) {
+            webview.focus();
```
Comment (akosyakov) [inline packages/plugin-ext/src/main/browser/webviews-main.ts]: `this.shell.activateWidget(widget)` - in order to tell the shell to bring a widget to focus, `webview` has to call `focus` in  `onActivateRequest` callback

**Comment by akosyakov** [comment]
@vparfonov How can one verify that it works? Do you have a VS Code extension?

**Comment by vparfonov** [comment]
This demo from master master
This from my branchreveal

I used this sample for testing https://github.com/Microsoft/vscode-extension-samples/tree/master/webview-sample

File: packages/plugin-ext/src/main/browser/webviews-main.ts (hunk 1 above)
Comment (akosyakov) [reply packages/plugin-ext/src/main/browser/webviews-main.ts]: also It looks bogus, `preserverFocus` means that a web view should not get focus

```ts
if (showOptions.preserveFocus) {
    shell.activateWidget(webview.id);
} else {
    shell.revealWidget(webview.id);
}
```

and then call  `this.node.focus()` in `onActivateRequest` callback on webview widget

I will modify sample extension to test it.

Comment (vparfonov) [reply packages/plugin-ext/src/main/browser/webviews-main.ts]: ok thanks

**Comment by akosyakov** [comment]
It does not seem to receive focus, if a widget has focus its tab should be highlighted blue. It happens automatically when any child node of a widget is `document.activeElement`.

**Comment by akosyakov** [comment]
https://github.com/theia-ide/theia/blob/48743de06e33ea812dddc3b70b0900ea9f77c754/packages/plugin-ext/src/main/browser/webviews-main.ts#L89-L92 is bogus as well, should be aligned.

I think `focus` function should not be exposed out of `Webview` and only used in `onActivateRequest`.

**Comment by vparfonov** [comment]
> is bogus as well, should be aligned.

Yes looks like you are right, i will remove this code. Thanks

**Comment by akosyakov** [comment]
I've modified the sample example with input field and it does not work. I will open an alternative PR to solve it.

**Pull request closed without merging**